### ISSUE TITLE
Usage with OpenTX 2.1

### ISSUE THREAD

**Issue by mr-sneezy** [projectkk2glider/APM-Mavlink-to-FrSky#5]
You have one of the most recently updated forks of the original project, so I thought you might still be interested in it.
I've been using the code on an Arduino Mini with an APM plane quite successfully for some while, but having now just updated my Taranis to OpenTX 2.1.3 I've found that many telemetry values are no longer working or read unexpected values.
For instance Temp 1 works and shows Mav mode, VFAS works,Heading works, GPS location works, Fuel works, and I think A1 & A2.
What doesn't work now is Altitude (reads a static number), Current, GPS Speed, GPS Alt (reads 10x too high I think).
Not sure if you use OpenTX 2.1 (which has some very nice telemetry features) but expect to have a few issues when you do if my experience is a guide.

I also use the telemetry on a Pixhawk which now has native FrSky telemetry, and all sensors read fine from that.

**Comment by projectkk2glider** [comment]
I am still on OpenTX 2.0, but plan on switching to 2.1 soon. I will look at this issues then. Thanks for reporting.

**Comment by mr-sneezy** [comment]
I like the changes in 2.1, I wouldn't go back now.

As an extra bit of info, today I re-setup a big glider I had not got of the shelf for a long while. It uses an FrSky Sensor Hub through a D8R receiver, and I have a GPS, Temp and A1 voltage divider set up on it, as well as it's own built in sensors. It works on Taranis OTX2.1 exactly like it should, all sensors report and report correctly, so OpenTX 2.1 telemetry does seem not to be an issue at all.

**Comment by projectkk2glider** [comment]
Just tested with OpenTX 2.1.4N494 and the sensors you report as not working:
- Alt (baro altitude) works for me
- Curr works 
- could not test GPS indoors

All other values also appear correct. So it seems there is some other problem on your side. One option perhaps is that you created/detected telemetry sensors in early OpenTX 2.1 and something has been changed since. Please make a backup of  your current model, then delete all telemetry sensors and re-detect them. Does that improve things? Do not forget to check any logical switches or other items that are dependent on telemetry values after you delete and re-detect sensors. Things might change/dissapear.

**Comment by mr-sneezy** [comment]
I guess you compiled that version of OpenTX yourself as I don't see an update after 2.1.3 yet.

I just tried what you suggested but went even further and created a new model entry and rebound it. 
Did a sensor discovery, and got the same results as described above again. 

I'll now try erasing the Arduino Mini sketch by loading Blink to it, then confirm that, then recompile your code again.
I used Arduino1.0.6 to compile your code as I got errors with 1.6.6. Is there any chance that could cause be a problem  like this ?

PS. I should have added in the OP that I simultaneously monitored the APM plane via a Mavlink radio while trying to fault find this, and all sensors read fine via that system...

**Comment by projectkk2glider** [comment]
I compiled mine with Arduino 1.6.3. The OpenTX is self compiled, but there is no relevant changes to the telemetry from the 2.1.3 version. Do you see the `*` characters appearing in the telemetry setup screen on the Taranis. Every time that symbol is displayed, the data for that sensor was received. So the \* should regularly appear before every sensor (some more frequently some less).

**Comment by projectkk2glider** [comment]
Did you do any other change beside the upgrade of OpenTX from 2.0 to 2.1? Like a configuration change of APM or firmware upgrade.

The static value of the baro altituede could be the result fo missconfiguration of the APM, ie it does not send it via Mavlink. See https://github.com/projectkk2glider/APM-Mavlink-to-FrSky/issues/4#issuecomment-128927454 for example setup.

The issue of GPS Alt (reads 10x too high I think) was solved in #1, are you sure you are using the latest release of this project? Use this https://github.com/projectkk2glider/APM-Mavlink-to-FrSky/archive/v1.0.zip and compile it, the included binaries are old!

**Comment by mr-sneezy** [comment]
Mystery solved, I suspect it was something I'd done after you reported no issues with OTX2.1.
I went back to basics with the 'Blink' test and it failed. IDE reported as 'Done Uploading' but the blink code clearly did not run. After trying a few things it was the board type in the IDE that was set wrong (Mini instead of Pro Mini) and once I got that right then Blink worked and I realized I never was running your code at all, it was still the old version from last year from who knows which fork :)
So the IDE reported a Done Upload but it was misleading me every time with the wrong board selected.

So, now I have the code from the link you sent above it does all seem to work fine like yours does. 
I've tried a copter Lua to see how things run and mostly that's right. I need to find a plane oriented LUA for 2.1 now or just setup the displays the old way.
Thanks for your help and suggestions.
I have a friend who will appreciate this too so I'll help him get a Pro Mini setup on his APM plane.
